package/gluon-mesh-babel/src/babel-respondd.c

```diff
@@ -0,0 +1,731 @@
+
+	}
+}
+
+
+static void readbabeldata(struct json_object *obj, void (*lineprocessor)(char*, struct json_object*))
+{
+	int sockfd ;
+
+	struct sockaddr_in6 serv_addr = {
+		.sin6_family = AF_INET6,
+		.sin6_port = htons(BABEL_PORT)
+	};
+
+	sockfd = socket(AF_INET6, SOCK_STREAM, 0);
+	if (sockfd < 0) {
+		perror("ERROR opening socket");
+		return;
+	}
+	if (inet_pton(AF_INET6, "::1", &serv_addr.sin6_addr.s6_addr) != 1)
+	{
+		perror("Cannot parse hostname");
+		return;
+	}
+	if (connect(sockfd, (const struct sockaddr *)&serv_addr, sizeof(serv_addr)) != 0) {
+		perror("Can not connect to babeld");
+		return;
+	}
+
+	// receive babel header
+	input_pump(sockfd, obj, lineprocessor);
+	int i =  write(sockfd, "dump\n", 5);
+	if ( i < 5 ) {
+		// TODO how to handle this?
+		printf("could not complete write \"dump\" on the babel socket\n");
+	}
+
+	shutdown(sockfd, SHUT_WR);
+	// receive result
+	input_pump(sockfd, obj, lineprocessor);
+
+	close(sockfd);
+	return;
+}
+
+
+static int get_all_client_count() {
+	size_t count=0;
+
+	FILE *f = fopen("/proc/net/ipv6_route", "r");
+	if (!f)
+		return -1;
+
+	char *line = NULL;
+	size_t len = 0;
+
+	struct json_object *site = gluonutil_load_site_config();
+	if (!site)
+		return -1;
+
+	struct json_object *ret = NULL;
+	if (!json_object_object_get_ex(site, "prefix6", &ret))
+		return -1;
+
+	char *prefix = (char*)json_object_get_string(ret);
+	if (!prefix)
+		return -1;
+
+	int ilen =strlen(prefix);
+	if (ilen > 3 )
+		prefix[ilen-3]='\0';
+	else
+		return -1;
+
+	struct in6_addr n_prefix;
+	inet_pton(AF_INET6, prefix, &n_prefix);
+
+	len = 0;
+	while (getline(&line, &len, f) >= 0) {
+		char ifname[IF_NAMESIZE+1];
+		struct in6_addr addr;
+
+		// scan lines, only consider lines matching site.prefix6 for the rest of the loop
+		if (sscanf(line,
+					"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8
+					"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8
+					" 80 %*32x %*2x %*32x %*8x %*8x %*8x %*8x %"STRINGIFY(IF_NAMESIZE)"s",
+					&addr.s6_addr[0], &addr.s6_addr[1], &addr.s6_addr[2], &addr.s6_addr[3],
+					&addr.s6_addr[4], &addr.s6_addr[5], &addr.s6_addr[6], &addr.s6_addr[7],
+					&addr.s6_addr[8], &addr.s6_addr[9], &addr.s6_addr[10], &addr.s6_addr[11],
+					&addr.s6_addr[12], &addr.s6_addr[13], &addr.s6_addr[14], &addr.s6_addr[15],
+					ifname) != 17)
+			continue;
+		if ((!strncmp(ifname, "local-node",10)) && (!memcmp(&n_prefix, &addr, 8)))
+			count++;
+	}
+	json_object_put(site);
+
+	fclose(f);
+
+	if (line)
+		free(line);
+
+	// exclude next-node-ip (it is always in the routing table), no need to really find it.
+	count--;
+
+	return count;
+}
+
+static struct json_object * get_clients(void) {
+	size_t wifi24 = 0, wifi5 = 0;
+
+	count_stations(&wifi24, &wifi5);
+
+	//	size_t total = atoi(get_line_from_run("exec sh -c 'ip -6 r s t 0 |grep 2a06|grep -v mesh-vpn|grep -v unreachable|grep -v /|wc -l'"));
+	size_t total = get_all_client_count();
```

Comment:
Additional: 
`data: {"clients":{"total":0,"wifi":1,"wifi24":1,"wifi5":0}`
Opposite seems to be true too, it might be that the wifi connection is still open, but the route might be gone?
Mobile phone was logged in for ~10 min, probably gone to sleep/energy save.
As soon as i switched the screen on, the info was suddenly correct:
`{"clients":{"total":1,"wifi":1,"wifi24":1,"wifi5":0}`